src/compiler/checker.ts

```diff
@@ -1295,6 +1295,12 @@ const enum MappedTypeModifiers {
     ExcludeOptional = 1 << 3,
 }
 
+const enum MappedTypeNameTypeKind {
+    None = 0,
+    Filtering = 1 << 0,
+    Remapping = 1 << 1,

```

Comment:
Why is this a flags enum? I can't see anywhere we actually combine multiple flags - in fact, it seems to me like a mapped type can only be one, given the implementation of `getMappedTypeNameTypeKind`.

Reply:
I had some composite values here before but removed them at the end since I didn't actually use them anywhere. Refactored this now to be a non-flag enum